activerecord/lib/active_record/persistence.rb

```diff
@@ -215,7 +215,7 @@ def becomes(klass)
       became.instance_variable_set("@changed_attributes", attributes_changed_by_setter)
       became.instance_variable_set("@new_record", new_record?)
       became.instance_variable_set("@destroyed", destroyed?)
-      became.instance_variable_set("@errors", errors)
+      became.errors.copy!(errors)
```

Comment:
`errors.dup` - not enough, because `errors.base` will have an old type of object, see:

```
  def test_becomes_errors_base
    child_class = Class.new(Admin::User) do
      store_accessor :settings, :foo # <== The child object has an extra field

      def self.name; 'Admin::ChildUser'; end
    end

    admin = Admin::User.new
    admin.errors.add :token, :invalid
    child = admin.becomes(child_class)

    assert_equal [:token], child.errors.keys
    assert_nothing_raised do
      child.errors.add :foo, :invalid # <== and here we will have a error...
    end
  end

  1) Error:
PersistenceTest#test_becomes_errors_base:
NoMethodError: undefined method `foo' for #<Admin::User:0x00000002167680>
    /vagrant/rails/activemodel/lib/active_model/attribute_methods.rb:433:in `method_missing'
    /vagrant/rails/activemodel/lib/active_model/errors.rb:474:in `generate_message'
    /vagrant/rails/activemodel/lib/active_model/errors.rb:490:in `normalize_message'
    /vagrant/rails/activemodel/lib/active_model/errors.rb:322:in `add'
    /vagrant/rails/activerecord/test/cases/persistence_test.rb:182:in `block in test_becomes_errors_base'
    /vagrant/rails/activesupport/lib/active_support/test_case.rb:85:in `assert_nothing_raised'
    /vagrant/rails/activerecord/test/cases/persistence_test.rb:181:in `test_becomes_errors_base'
```